holoviews/plotting/util.py

```diff
@@ -1330,7 +1330,11 @@ def _process(self, element, key=None):
             return
         column = agg.column
         if hasattr(hvds.data, 'dtypes'):
-            cats = list(hvds.data.dtypes[column].categories)
+            try:
+                cats = list(hvds.data.dtypes[column].categories)
+            except TypeError:
+                # Issue #5619, cudf.core.index.StringIndex is not iterable.
+                cats = list(hvds.data.dtypes[column].categories.values_host)
```

Comment:
```suggestion
                cats = list(hvds.data.dtypes[column].categories.to_pandas())
```

I like this more because it could be used for other DataFrame libraries if needed. An example is `polars` which have the same method.
https://pola-rs.github.io/polars/py-polars/html/reference/dataframe/api/polars.DataFrame.to_pandas.html

Reply:
Here we are dealing with some `Index` class rather than a `DataFrame` or `Series`, and I don't know what `polars` uses for indexes. But yes, it is a good idea to standardise.